src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,26 @@
+import Popup from "./Popup.js";
+class PopupWithConfirm extends Popup {
+  constructor(popupSelector, handleConfirmDelete) {
+    super({ popupSelector });
+    this._popupButton = this._popupElement.querySelector(".modal__button");
+    this._handleConfirmDelete = handleConfirmDelete;
+    this._card;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._popupButton.addEventListener("click", (evt) => {
+      this._handleConfirmDelete(this._card);
+    });
+  }
+
+  open(card) {
+    super.open();
+    this._card = card;
+  }
+
+  close() {
+    super.close();
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.